docs/advanced/automated-testing.md

```diff
@@ -0,0 +1,262 @@
+
+Now we can move on to the next topic
+
+## Automating the testing
+
+For this we’re going to use the free tools provided by GitHub to every public(and most private?) repository, this assumes you already have a git repository hosted on GitHub, if you need help doing that I can point you to their exhaustive docs:
+
+https://docs.github.com/en/get-started/quickstart
+
+And also the documentation about actions specifically:
+
+https://docs.github.com/en/actions
+
+So, first we’re going to create a workflow file
+
+Set up a folder structure like so:
+
+```
+Root project folder/
+  .github/
+    workflows/
+      testing.yml
+```
+
+```yaml title=testing.yml
+name: TSTL Testing
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  build_tests:
+    name: Busted Tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js [16.x]
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16.x
+          cache: "npm"
+      - name: Npm Install && Build with Testing preset
+        run: npm install && npm run-script build
+      - name: Install Lua
+        uses: leafo/gh-actions-lua@v8
+        with:
+          luaVersion: "luajit-2.1.0-beta3"
+      - name: Install LuaRocks
+        uses: leafo/gh-actions-luarocks@v4.0.0
+      - name: Install Busted
+        run: luarocks install busted
+      - name: Running Tests
+        run: cd dist && busted
+```
+
+Without going in to much detail, this workflow triggers whenever you push a commit to your github repository, installs Node.js, Lua, Luarocks and Busted installs all of your project dependencies and compiles a fresh version of the project, at the very end it moves in to the dist folder and runs your busted tests exactly like you would. Now, if you commit and push these changes to your GitHub repository you should be able to see, Under the Actions tab at the top, something like this:
+
+![Github action result](/images/github-workflow-result.png)
+
+Now that’s just it, this job will run every time you push a new commit to the repository, if it fails it’ll show up as a red cross, and you can find which test failed in the logs.
+
+## TSTL Library specifics
+
+This section applies to projects described in [Publishing Modules](publishing-modules.md), as you might know, library projects don’t (and shouldn’t) include the luaLibBundle and also any dependencies it would need to run, so how do we make it run standalone, for testing, we’re going to utilize the fact that you can pass in a different `tsconfig.json` to the compiler, so, the regular one is implicitly used based on the name `tsconfig.json`, we’re going to create a new one like `tsconfig.test.json` you can pretty much duplicate it from your current one, we’re just going to change a few values:
```

Comment:
Why are you suddenly mentioning luaLibBundle here? this is the first time that's referenced in this article, and I don't see how it is relevant - shouldn't be needed for busted right?

Reply:
This is a weird setup, why not have a tsconfig.json inside your /spec/ directory? It should also be possible to just give that its own dist (or just do no outdir and just have lua next to TS files in that directory).